### PR TITLE
Add mark_runs and compress_runs

### PR DESCRIPTION
**Pull request by nvictus**
I personally use this all the time. Figured it's finally time to add it.

### CODE REVIEW

**Comment by nvictus** [comment]
Question: which name do we like better, `compress_runs` or `merge_runs`?

It basically very similar to `merge`, but instead of combining all intervals which transitively overlap (thus, sharing a unique cluster ID), it takes the hull of intervals that are (1) back-to-back and (2) share a value in a column (thus, sharing a unique run ID).

**Comment by nvictus** [comment]
Now I'm wondering if these couldn't just be special cases of `cluster` and `merge`.

File: bioframe/extras.py [hunk 1]
```diff
@@ -499,3 +503,161 @@ def pair_by_distance(
...
+    1   chr1    300  600      2    6
+
+    See Also
+    --------
+    mark_runs
+    """
+    ck, sk, ek = _get_default_colnames() if cols is None else cols
+
+    if agg is None:
+        agg = {}
```
Comment (GarrettNg) [inline bioframe/extras.py:643]: This could be set as a default parameter

**Review by GarrettNg — APPROVED** [review]

File: bioframe/extras.py (hunk 1 above)
Comment (nvictus) [reply bioframe/extras.py:643]: Mutable defaults in Python are an [antipattern](https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/)

Comment (GarrettNg) [reply bioframe/extras.py:643]: Interesting... wonder if that's by language design :sweat_smile:

**Pull request merged**